File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
int maxcount=0;
        for(int i=0;i<nums.length;i++){
            int zerocount=0;
            int tempcount=0;
            for(int j=i;j<nums.length;j++){
                if(nums[j]==1){
                tempcount++;              
            }
            else if(nums[j]==0&&zerocount<k){
                zerocount++;
                tempcount++;
            }
            else break;
        }
             maxcount=Math.max(maxcount,tempcount);
        }
        return maxcount;
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int zerocount=k;
        int start=0;
        int maxsize=0;
        int curr=0;
        while(curr<nums.size()){
            if(nums[curr]==0&&zerocount==0){
                while(true){
                    if(nums[start]==0){
                         start++;
                        zerocount++;
                        break;
                     }
                start++;
                }
                zerocount--; 
            }
            else if(nums[curr]==0){
                zerocount--;
            }
            maxsize=max(maxsize,(curr-start)+1);
            curr++;
        }
        return maxsize;
    }
};